src/briefcase/commands/new.py

```diff
@@ -103,7 +106,7 @@ def validate_app_name(self, candidate):
         :returns: True. If there are any validation problems, raises ValueError
             with a diagnostic message.
         """
-        if not PEP508_NAME_RE.match(candidate):
+        if not PEP508_NAME_RE.match(candidate) and candidate.isascii():  # prevent error for non-ascii characters
```

Comment:
This indicates that we're not using the PEP508_NAME_RE correctly - the definition comes directly from PEP508, and we're not telling the regex to match ASCII specifically, so it will match unicode as well. Passing `re.ASCII` as a flag when compiling the regex should address this.